src/animation.js

```diff
@@ -55,17 +56,74 @@ class CompositeAnimation {
 	}
 }
 
-let WalkAnimation = (player, time) => {
+let WalkingAnimation = (player, time) => {
 	let skin = player.skin;
-	let angleRot = time + Math.PI / 2;
 
-	// Leg Swing
-	skin.leftLeg.rotation.x = Math.cos(angleRot);
-	skin.rightLeg.rotation.x = Math.cos(angleRot + (Math.PI));
+	// Leg swing
+	skin.leftLeg.rotation.x = Math.sin(time) * 0.3;
+	skin.rightLeg.rotation.x = Math.sin(time + Math.PI) * 0.3;
 
-	// Arm Swing
-	skin.leftArm.rotation.x = Math.cos(angleRot + (Math.PI));
-	skin.rightArm.rotation.x = Math.cos(angleRot);
+	// Arm swing
+	skin.leftArm.rotation.x = Math.sin(time + Math.PI) * 0.5;
+	skin.rightArm.rotation.x = Math.sin(time) * 0.5;
+	let basicArmRotationZ = Math.PI * 0.02;
+	skin.leftArm.rotation.z = Math.cos(time) * 0.03 + basicArmRotationZ;
+	skin.rightArm.rotation.z = Math.cos(time + Math.PI) * 0.03 - basicArmRotationZ;
+
+	// Head shaking with different frequency & amplitude
+	skin.head.rotation.y = Math.sin(time / 2) * 0.2;
+	skin.head.rotation.x = Math.sin(time / 3) * 0.15;
+
+	// Always add an angle for cape around the x axis
+	let basicCapeRotationX = Math.PI * 0.06;
+	player.cape.rotation.x = Math.sin(time / 1.5) * 0.06 + basicCapeRotationX;
+};
+
+WalkingAnimation.naturalSpeed = 5;
+
+let RunningAnimation = (player, time) => {
+	let skin = player.skin;
+
+	// Leg swing with larger amplitude
+	skin.leftLeg.rotation.x = Math.cos(time + Math.PI) * 1.3;
+	skin.rightLeg.rotation.x = Math.cos(time) * 1.3;
+
+	// Arm swing
+	skin.leftArm.rotation.x = Math.cos(time) * 1.5;
+	skin.rightArm.rotation.x = Math.cos(time + Math.PI) * 1.5;
+	let basicArmRotationZ = Math.PI * 0.1;
+	skin.leftArm.rotation.z = Math.cos(time) * 0.1 + basicArmRotationZ;
+	skin.rightArm.rotation.z = Math.cos(time + Math.PI) * 0.1 - basicArmRotationZ;
+
+	// Jumping
+	player.position.y = Math.cos(time * 2);
+	// Dodging when running
+	player.position.x = Math.cos(time) * 0.15;
+	// Slightly tilting when running
+	player.rotation.z = Math.cos(time + Math.PI) * 0.01;
+
+	// Apply higher swing frequency, lower amplitude,
+	// and greater basic rotation around x axis,
+	// to cape when running.
+	let basicCapeRotationX = Math.PI * 0.3;
+	player.cape.rotation.x = Math.sin(time * 2) * 0.1 + basicCapeRotationX;
+
+	// What about head shaking?
+	// You shouldn't glance right and left when running dude :P
+};
+
+RunningAnimation.naturalSpeed = 13;
+
+let RotatingAnimation = (player, time) => {
```

Comment:
I don't think it's necessary, `OrbitControls` has [an auto-rotating feature](https://github.com/to2mbn/skinview3d/blob/a654cd20e1ad5d65e442d628e5c00cc7c06bc9f9/src/orbit_controls.js#L94). It rotates the camera instead of the object.

However it requires `.update()` to be called every animation loop.

Reply:
Why not just leave it to animation functions? You are separating the ability of animating `PlayerObject` into two different parts.

Just let animation functions to animate (rotate) the `PlayerObject`, and let the `OrbitControls` just do its controling works (moving camera).